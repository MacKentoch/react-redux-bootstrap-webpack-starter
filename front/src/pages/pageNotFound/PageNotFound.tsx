import React from 'react';
import { RouteComponentProps, withRouter } from 'react-router';
import Jumbotron from 'reactstrap/lib/Jumbotron';
import FadeInEntrance from '../../components/fadeInEntrance';
import { OwnProps, ReduxConnectedProps } from './index';

// #region types
export type Props = RouteComponentProps & ReduxConnectedProps & OwnProps;
// #endregion

function PageNotFound() {
  return (
    <FadeInEntrance>
      <Jumbotron>
        <h1>Sorry this page does not exists...</h1>
      </Jumbotron>
    </FadeInEntrance>
  );
}

PageNotFound.displayName = 'PageNotFound';

export default withRouter(PageNotFound);
