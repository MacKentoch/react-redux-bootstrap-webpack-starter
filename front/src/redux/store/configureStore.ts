import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { createBrowserHistory as createHistory } from 'history';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import reducer from '../modules/reducers';

// #region constants
const isProd = process.env.NODE_ENV === 'production';
export const history = createHistory();
// #endregion

// #region createStore : enhancer

// #region logger middleware (dev only)
const loggerMiddleware = createLogger({
  level: 'info',
  collapsed: true,
});
// #endregion

const enhancer = !isProd
  ? composeWithDevTools(
      applyMiddleware(
        thunkMiddleware,
        loggerMiddleware, // logger at the end
      ),
    )
  : composeWithDevTools(applyMiddleware(thunkMiddleware));
// #endregion

// #region persisted reducer
const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['router'],
  // whitelist: ['userAuth'],
};

const persistedReducer = persistReducer(persistConfig, reducer);
// #endregion

export default function configureStore(initialState = {}) {
  const store = createStore(persistedReducer, initialState, enhancer);
  const persistor = persistStore(store);

  if ((module as any).hot) {
    (module as any).hot?.accept('../modules/reducers', () => {
      // @ts-ignore
      const nextAppReducer = require('../modules/reducers').default;
      store.replaceReducer(persistReducer(persistConfig, nextAppReducer));
    });
  }

  return { store, persistor };
}
