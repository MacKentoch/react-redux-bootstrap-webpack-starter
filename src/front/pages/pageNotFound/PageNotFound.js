// @flow

// #region lifecycle
import React, { PureComponent } from 'react';
import {
  type Match,
  type Location,
  type RouterHistory,
} from 'react-router-dom';
import Jumbotron from 'reactstrap/lib/Jumbotron';
// #endregion

// #region flow types
type Props = {
  // react-router 4:
  match: Match,
  location: Location,
  history: RouterHistory,

  ...any,
};
type State = any;
// #endregion

class PageNotFound extends PureComponent<Props, State> {
  // #region lifecycle
  render() {
    return (
      <div>
        <Jumbotron>
          <h1>Sorry this page does not exists...</h1>
        </Jumbotron>
      </div>
    );
  }
  // #endregion
}

export default PageNotFound;
