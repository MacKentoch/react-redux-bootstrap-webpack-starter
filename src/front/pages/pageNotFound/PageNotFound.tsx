import React from 'react';
import { RouteComponentProps } from 'react-router';
import Jumbotron from 'reactstrap/lib/Jumbotron';
import FadeInEntrance from '../../components/fadeInEntrance';

// #region types
type Props = {} & RouteComponentProps;
// #endregion

function PageNotFound(props: Props) {
  return (
    <FadeInEntrance>
      <Jumbotron>
        <h1>Sorry this page does not exists...</h1>
      </Jumbotron>
    </FadeInEntrance>
  );
}

PageNotFound.displayName = 'PageNotFound';

export default PageNotFound;
