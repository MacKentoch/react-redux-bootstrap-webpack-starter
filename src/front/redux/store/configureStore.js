// @flow

/* eslint no-process-env:0 */
if (process.env.NODE_ENV === 'production') {
  module.exports = require('./configureStore.prod');
} else {
  module.exports = require('./configureStore.dev');
}
