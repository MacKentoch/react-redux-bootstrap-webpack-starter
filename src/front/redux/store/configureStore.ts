/* eslint no-process-env:0 */
let configureStore = null;

if (process.env.NODE_ENV === 'production') {
  configureStore = require('./configureStore.prod').default;
  // const {history} = require('./configureStore.prod');
  // export history;
} else {
  configureStore = require('./configureStore.dev').default;
  // const {history} = require('./configureStore.dev');
  // export history;
}
export default configureStore;
