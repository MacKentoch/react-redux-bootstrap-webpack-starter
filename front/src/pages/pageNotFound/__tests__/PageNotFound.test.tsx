import React from 'react';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import PageNotFound from '../index';

const middlewares: Array<any> = [];
const mockStore = configureStore(middlewares);

describe('PageNotFound page', () => {
  let rootElement: any = null;

  beforeEach(() => {
    rootElement = document.createElement('div');
    document.body.appendChild(rootElement);
  });

  afterEach(() => {
    rootElement && document.body.removeChild(rootElement);
    rootElement = null;
  });

  it('renders as expected', () => {
    const initialState = {};
    const store = mockStore(initialState);

    const { container } = render(
      <Provider store={store}>
        <ThemeProvider theme={{}}>
          <MemoryRouter>
            <PageNotFound />
          </MemoryRouter>
        </ThemeProvider>
      </Provider>,
      rootElement,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
