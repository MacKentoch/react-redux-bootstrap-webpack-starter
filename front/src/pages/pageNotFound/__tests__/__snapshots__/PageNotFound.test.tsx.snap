// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound page renders as expected 1`] = `
<div
  class="sc-bdVaJa jxVqOJ"
>
  <div
    class="jumbotron"
  >
    <h1>
      Sorry this page does not exists...
    </h1>
  </div>
</div>
`;
